src/common/webrequest.cpp

```diff
@@ -0,0 +1,485 @@
+        }
+        else
+            m_dataSize = dataSize;
+    }
+    else
+        m_dataSize = 0;
+
+    SetHeader("Content-Type", contentType);
+}
+
+wxFileOffset wxWebRequest::GetBytesReceived() const
+{
+    return m_bytesReceived;;
+}
+
+wxFileOffset wxWebRequest::GetBytesExpectedToReceive() const
+{
+    if ( GetResponse() )
+        return GetResponse()->GetContentLength();
+    else
+        return -1;
+}
+
+void wxWebRequest::SetState(State state, const wxString & failMsg)
+{
+    // Add a reference while the request is active
+    if ( IsActiveState(state) && !IsActiveState(m_state) )
+        IncRef();
+
+    m_state = state;
+
+    // Trigger the event in the main thread
+    CallAfter(&wxWebRequest::ProcessStateEvent, state, failMsg);
+}
+
+void wxWebRequest::ReportDataReceived(size_t sizeReceived)
+{
+    m_bytesReceived += sizeReceived;
+}
+
+// The SplitParamaters implementation is adapted to wxWidgets
+// from Poco::Net::MessageHeader::splitParameters
+
+void wxWebRequest::SplitParameters(const wxString& s, wxString& value,
+    wxWebRequestHeaderMap& parameters)
+{
+    value.clear();
+    parameters.clear();
+    wxString::const_iterator it = s.begin();
+    wxString::const_iterator end = s.end();
+    while ( it != end && wxIsspace(*it) ) ++it;
+    while ( it != end && *it != ';' ) value += *it++;
+    value.Trim();
+    if ( it != end ) ++it;
+    SplitParameters(it, end, parameters);
+}
+
+void wxWebRequest::SplitParameters(const wxString::const_iterator& begin,
+    const wxString::const_iterator& end, wxWebRequestHeaderMap& parameters)
+{
+    wxString pname;
+    wxString pvalue;
+    pname.reserve(32);
+    pvalue.reserve(64);
+    wxString::const_iterator it = begin;
+    while ( it != end )
+    {
+        pname.clear();
+        pvalue.clear();
+        while ( it != end && wxIsspace(*it) ) ++it;
+        while ( it != end && *it != '=' && *it != ';' ) pname += *it++;
+        pname.Trim();
+        if ( it != end && *it != ';' ) ++it;
+        while ( it != end && wxIsspace(*it) ) ++it;
+        while ( it != end && *it != ';' )
+        {
+            if ( *it == '"' )
+            {
+                ++it;
+                while ( it != end && *it != '"' )
+                {
+                    if ( *it == '\\' )
+                    {
+                        ++it;
+                        if ( it != end ) pvalue += *it++;
+                    }
+                    else pvalue += *it++;
+                }
+                if ( it != end ) ++it;
+            }
+            else if ( *it == '\\' )
+            {
+                ++it;
+                if ( it != end ) pvalue += *it++;
+            }
+            else pvalue += *it++;
+        }
+        pvalue.Trim();
+        if ( !pname.empty() ) parameters[pname] = pvalue;
+        if ( it != end ) ++it;
+    }
+}
+
+void wxWebRequest::ProcessStateEvent(State state, const wxString& failMsg)
+{
+    if (!IsActiveState(state) && GetResponse())
```

Comment:
```suggestion
    if ( !IsActiveState(state) && GetResponse() )
```

Just for consistency.